python/ray/dag/compiled_dag_node.py

```diff
@@ -380,6 +367,10 @@ def __init__(
         # Type hints specified by the user for DAG (intermediate) outputs.
         self._type_hints = []
 
+        # Uniquely identifies the _NCCLGroup that will be used within this DAG,
```

Comment:
What is NCCLGroup?

Reply:
It represents a [NCCL communicator](https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/usage/communicators.html#).